lib/recipe_cli/cli.rb

```diff
@@ -0,0 +1,46 @@
+
+class Cli 
+
+    class TypeError < StandardError
+        def message
+           puts "Unfortunately, you have selected a category that does not exist. Please select from one of the available 11 categories." 
+        end 
+      end  
+      
+      class TypeError < StandardError
+        def message
+           puts "Unfortunately, you have selected the number of a recipe that does not exist. Please select from the available options."  
+        end 
+      end  
+      
+
+    def greeting
+        puts "Welcome to TheMealDB, please enter in your first name."
+        user_input = gets.chomp
+        puts "Hello, Chef #{user_input.capitalize}! Today you will be able to find a delicious recipe." 
+        puts "To get started, enter the number of the meal category."
+        user_input = gets.chomp
```

Comment:
[SYNTAX/STYLE]

You've got `user_input` defined multiple times here, presumably to get user input multiple times. It works, but it's potentially a source of bugs further down the line if you want to get hold of some user input from earlier on but it's been overwritten. I'd suggest using a new variable name for each user input you want:

```suggestion
        chef_name = gets.chomp
        puts "Hello, Chef #{user_input.capitalize}! Today you will be able to find a delicious recipe." 
        puts "To get started, enter the number of the meal category."
        category_input = gets.chomp
```

There's more than just these two, so you'll need to re-read through your method and find each time you're getting more input and give them a new descriptive name instead.